modules/gapi/include/opencv2/gapi/streaming/onevpl/data_provider_interface.hpp

```diff
@@ -41,22 +56,32 @@ struct GAPI_EXPORTS DataProviderSystemErrorException : public DataProviderExcept
  */
 struct GAPI_EXPORTS IDataProvider {
     using Ptr = std::shared_ptr<IDataProvider>;
+    using mfx_codec_id_type = uint32_t;
+
+    virtual ~IDataProvider() = default;
 
-    virtual ~IDataProvider() {}
+    /**
+     * The function is used by onevpl::GSource to extract codec id from data
+     *
+     */
+    virtual mfx_codec_id_type get_mfx_codec_id() const = 0;
 
     /**
      * The function is used by onevpl::GSource to extract binary data stream from @ref IDataProvider
      * implementation.
      *
      * It MUST throw `DataProviderException` kind exceptions in fail cases.
-     * It MUST return 0 in EOF which considered as not-fail case.
+     * It MUST return MFX_ERR_MORE_DATA in EOF which considered as not-fail case.
      *
      * @param out_data_bytes_size the available capacity of out_data buffer.
-     * @param out_data the output consumer buffer with capacity out_data_bytes_size.
+     * @param in_out_bitsream the input-output reference on MFX bitstream buffer which MUST be empty at the first request
+     * to allow implementation to allocate it by itself and to return back. Subsequent invocation of `fetch_bitstream_data`
+     * MUST use the previously used in_out_bitsream to avoid skipping rest of hasn't been consumed frames
      * @return fetched bytes count.
      */
-    virtual size_t fetch_data(size_t out_data_bytes_size, void* out_data) = 0;
-
+#ifdef HAVE_ONEVPL
```

Comment:
I don't think that using `ifdef`s in a public header works. Please, remove the guard and handle it in the sources

Reply:
Have you added `#else` with assert? I think other reviewers need to check on that